projects/igniteui-angular/src/lib/core/selection.ts

```diff
@@ -19,7 +19,9 @@ export class IgxSelectionAPIService {
      * @param newSelection The new component selection to be set.
      */
     public set(componentID: string, newSelection: Set<any>) {
-        this.selection.set(componentID, newSelection);
+        if (componentID) {
```

Comment:
Shouldn't there be an exception if the component ID is `undefined`? To me this scenario is not valid.